src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -15,640 +15,6 @@ namespace Microsoft.CodeAnalysis.CSharp.Extensions
-                    // Cast removal can change program behavior.                    
-                    return !(expressionToCastType.IsNumeric && castToOuterType.IsNumeric);
-                }
-
-                // Case :
-                // 2. object y = (ValueType)1;
-                if (expressionToCastType.IsBoxing && expressionToOuterType.IsBoxing && castToOuterType.IsImplicit)
-                {
-                    return true;
-                }
-
-                // Case :
-                // 3. object y = (NullableValueType)null;
-                if ((!castToOuterType.IsBoxing || expressionToCastType.IsNullLiteral) &&
-                    castToOuterType.IsImplicit &&
-                    expressionToCastType.IsImplicit &&
-                    expressionToOuterType.IsImplicit)
-                {
-                    if (expressionToOuterType.IsAnonymousFunction)
-                    {
-                        return expressionToCastType.IsAnonymousFunction &&
-                            !speculationAnalyzer.ReplacementChangesSemanticsOfUnchangedLambda(cast.Expression, speculationAnalyzer.ReplacedExpression);
-                    }
-
-                    return true;
-                }
-            }
-
-            return false;
-        }
-
-        private static bool IsRequiredImplicitNumericConversion(ITypeSymbol sourceType, ITypeSymbol destinationType)
-        {
-            // C# Language Specification: Section 6.1.2 Implicit numeric conversions
-
-            // Conversions from int, uint, long, or ulong to float and from long or ulong to double may cause a loss of precision,
-            // but will never cause a loss of magnitude. The other implicit numeric conversions never lose any information.
-
-            switch (destinationType.SpecialType)
-            {
-                case SpecialType.System_Single:
-                    switch (sourceType.SpecialType)
-                    {
-                        case SpecialType.System_Int32:
-                        case SpecialType.System_UInt32:
-                        case SpecialType.System_Int64:
-                        case SpecialType.System_UInt64:
-                            return true;
-
-                        default:
-                            return false;
-                    }
-
-                case SpecialType.System_Double:
-                    switch (sourceType.SpecialType)
-                    {
-                        case SpecialType.System_Int64:
-                        case SpecialType.System_UInt64:
-                            return true;
-
-                        default:
-                            return false;
-                    }
-
-                default:
-                    return false;
-            }
-        }
-
         public static ExpressionSyntax Uncast(this CastExpressionSyntax node)
```

Comment:
This should probably also move to CastSimplifier. After all it's the actual bit which does the cast simplifying! And it doesn't look like it should be an extension method.

Reply:
I'm ok keeping this here.  This does feel much more like an simple syntax extension.  What i'm trying to do is keep the complex simplification-extensions centralized as they tend to be a ton of intertwined code that i don't want mixed in with lots of small independent helper extensions.